integration_test/reporting_error_index/reporting_error_index_test.go

```diff
@@ -0,0 +1,814 @@
+	config.Set("CONFIG_BACKEND_URL", cbURL)
+	config.Set("WORKSPACE_TOKEN", "token")
+	config.Set("DB.port", postgresContainer.Port)
+	config.Set("DB.user", postgresContainer.User)
+	config.Set("DB.name", postgresContainer.Database)
+	config.Set("DB.password", postgresContainer.Password)
+	config.Set("DEST_TRANSFORM_URL", transformerURL)
+
+	config.Set("Warehouse.mode", "off")
+	config.Set("DestinationDebugger.disableEventDeliveryStatusUploads", true)
+	config.Set("SourceDebugger.disableEventUploads", true)
+	config.Set("TransformationDebugger.disableTransformationStatusUploads", true)
+	config.Set("JobsDB.backup.enabled", false)
+	config.Set("JobsDB.migrateDSLoopSleepDuration", "60m")
+	config.Set("archival.Enabled", false)
+	config.Set("Reporting.syncer.enabled", false)
+	config.Set("Reporting.errorIndexReporting.enabled", true)
+	config.Set("Reporting.errorIndexReporting.syncer.enabled", true)
+	config.Set("Reporting.errorIndexReporting.SleepDuration", "1s")
+	config.Set("Reporting.errorIndexReporting.minWorkerSleep", "1s")
+	config.Set("Reporting.errorIndexReporting.uploadFrequency", "1s")
+	config.Set("BatchRouter.mainLoopFreq", "1s")
+	config.Set("BatchRouter.uploadFreq", "1s")
+	config.Set("Gateway.webPort", strconv.Itoa(port))
+	config.Set("RUDDER_TMPDIR", os.TempDir())
+	config.Set("recovery.storagePath", path.Join(tmpDir, "/recovery_data.json"))
+	config.Set("recovery.enabled", false)
+	config.Set("Profiler.Enabled", false)
+	config.Set("Gateway.enableSuppressUserFeature", false)
+
+	config.Set("ErrorIndex.storage.Bucket", minioResource.BucketName)
+	config.Set("ErrorIndex.storage.Endpoint", minioResource.Endpoint)
+	config.Set("ErrorIndex.storage.AccessKey", minioResource.AccessKey)
+	config.Set("ErrorIndex.storage.SecretAccessKey", minioResource.SecretKey)
+	config.Set("ErrorIndex.storage.S3ForcePathStyle", true)
+	config.Set("ErrorIndex.storage.DisableSSL", true)
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panicked: %v", r)
+		}
+	}()
+	r := runner.New(runner.ReleaseInfo{EnterpriseToken: "DUMMY"})
+	c := r.Run(ctx, []string{"rudder-error-reporting"})
+	if c != 0 {
+		err = fmt.Errorf("rudder-server exited with a non-0 exit code: %d", c)
+	}
+	return
+}
+
+func requireJobsCount(t *testing.T, db *sql.DB, queue, state string, expectedCount int) { // nolint: unparam
+	t.Helper()
+
+	require.Eventually(t, func() bool {
+		var jobsCount int
+		require.NoError(t, db.QueryRow(fmt.Sprintf(`SELECT count(*) FROM unionjobsdbmetadata('%s',1) WHERE job_state = '%s';`, queue, state)).Scan(&jobsCount))
+		t.Logf("%s %sJobCount: %d", queue, state, jobsCount)
+		return jobsCount == expectedCount
+	},
+		20*time.Second,
+		1*time.Second,
+		fmt.Sprintf("%d %s events should be in %s state", expectedCount, queue, state),
+	)
+}
+
+func requireMessagesCount(t *testing.T, ctx context.Context, mr *destination.MINIOResource, expectedCount int) { // nolint: unparam
```

Comment:
We can add some additional parameters to be passed as query criteria, e.g.
- failed_stage
- source_id
- destination_id
- event_type